capa/features/extractors/cape/call.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at: [package root]/LICENSE.txt
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+import logging
+from typing import Any, Dict, List, Tuple, Iterator
+
+import capa.features.extractors.cape.file
+import capa.features.extractors.cape.thread
+import capa.features.extractors.cape.global_
+import capa.features.extractors.cape.process
+from capa.features.insn import API, Number
+from capa.features.common import String, Feature
+from capa.features.address import Address, DynamicReturnAddress
+from capa.features.extractors.base_extractor import CallHandle, ThreadHandle, ProcessHandle
+
+logger = logging.getLogger(__name__)
+
+
+def extract_call_features(
+    behavior: Dict, ph: ProcessHandle, th: ThreadHandle, ch: CallHandle
+) -> Iterator[Tuple[Feature, Address]]:
+    """
+    this method extrcts the given call's features (api name and arguments),
+    and returns them as API, Number, and String features.
+
+    args:
+      behavior: a dictionary of behavioral artifacts extracted by the sandbox
+      ph: process handle (for defining the extraction scope)
+      th: thread handle (for defining the extraction scope)
+      ch: call handle (for defining the extraction scope)
+
+    yields:
+      Feature, address; where Feature is either: API, Number, or String.
+    """
+    # TODO(yelhamer): find correct base address used at runtime.
+    # this address may vary from the PE header, may read actual base from procdump.pe.imagebase or similar.
+    # https://github.com/mandiant/capa/issues/1618
+    process = capa.features.extractors.cape.helpers.find_process(behavior["processes"], ph)
+    calls: List[Dict[str, Any]] = process["calls"]
+    call = calls[ch.address.id]
+    assert call["thread_id"] == str(th.address.tid)
+    caller = DynamicReturnAddress(call=ch.address, return_address=int(call["caller"], 16))
+    # list similar to disassembly: arguments right-to-left, call
+    for arg in call["arguments"][::-1]:
+        try:
+            yield Number(int(arg["value"], 16), description=f"{arg['name']}"), caller
+        except ValueError:
+            yield String(arg["value"], description=f"{arg['name']}"), caller
```

Comment:
```suggestion
            yield Number(int(arg["value"], 16)), caller
        except ValueError:
            yield String(arg["value"]), caller
```

clever use of the `description` kwarg, but today its only meant for rule authors. we could think about extending its interpretation like you do here, but i think that should be a separate discussion.